app/routes/_gcn.circulars/circulars.server.ts

```diff
@@ -173,12 +201,30 @@ export async function search({
 }
 
 /** Get a circular by ID. */
-export async function get(circularId: number): Promise<Circular> {
+export async function get(
+  circularId: number,
+  version?: number
+): Promise<Circular> {
   if (isNaN(circularId)) throw new Response(null, { status: 404 })
   const db = await tables()
-  const result = await db.circulars.get({
-    circularId,
-  })
+
+  let result: Circular
+
+  if (version) {
+    if (isNaN(version)) throw new Response(null, { status: 404 })
+    result = await db.circulars_history.get({
+      circularId,
+      version,
+    })
+    if (!result)
+      throw new Response('Specified version does not exist for this circular', {
+        status: 404,
+      })
+  } else {
+    result = await db.circulars.get({
+      circularId,
+    })
+  }
```

Comment:
A couple issues with this:

- The code within the `if (version) { if (isNaN(version)) { ... }}` block cannot be reached because `NaN` is falsey.
- When getting a Circular by version, we _still_ need to handle the case that the current version _is_ the requested version.
- The new `if (!result)` section is redundant because that check already exists below.

One possible refactoring:

```suggestion
  if (version) {
    result = await db.circulars_history.get({
      circularId,
      version,
    })
  }
  if (!result && !isNaN(version)) {
    result = await db.circulars.get({
      circularId
    })
  }
  if (version && result?.version !== version)
  {
    result = undefined
  }
```